src/os/CMakeLists.txt

```diff
@@ -58,6 +58,7 @@ if(HAVE_LIBZFS)
 endif()
 
 add_library(os STATIC ${libos_srcs})
+target_link_libraries(os blkdrv)
```

Comment:
ideally, all the commits should be atomic. so why do we need to link os against blkdrv?

Reply:
Just follow the same usage in src/os/CMakeLists.txt and don't want to change it in case of raising some unknown problem:
https://github.com/ceph/ceph/blob/master/src/os/CMakeLists.txt#L92
https://github.com/ceph/ceph/blob/master/src/os/CMakeLists.txt#L104
https://github.com/ceph/ceph/blob/master/src/os/CMakeLists.txt#L129
https://github.com/ceph/ceph/blob/master/src/os/CMakeLists.txt#L143

Reply:
@changchengx my question is regarding to the commit message

> link library libblk into the dependent module

it does not answer the question of "why do we need this change".

Reply:
@tchaikov 
Do you mean that "every commit" should pass build and keep Ceph work?
If yes, I also have this concern. If we follow this rule, I need to squash the commits in this PR into one commit.

Reply:
> @tchaikov
> Do you mean that "every commit" should pass build and keep Ceph work?

ideally, yes. we have exceptions though, in case where squashed commits render it very difficult to understand / review the changes. for instance, in #29113, i split the PR into 65 commits, quite a few of them cannot compile. but if i don't break them down into smaller pieces, the reviews will be lost.

> If yes, I also have this concern. If we follow this rule, I need to squash the commits in this PR into one commit.

also, again, the commit message fails not explain the "why". i think it is as important as "what".

Reply:
@tchaikov 
I've refined the commit message.
If it's needed, I can squash the two commits into one after all the discussions have been solved in this PR.

Reply:
Keep "atomic" now.